File: java/src/sample/replicant/sample2/MockWidgetPainter.java
// Copyright 2009 Kiel Hodges
package replicant.sample2;

import replicant.*;
import static replicant.Replicant.*;

public class MockWidgetPainter implements WidgetPainter {

  public MockWidgetPainter(ExpectationEnforcer expectationEnforcer) {
    paint = voidMethodMocker(this, "paint").
      mappingResponses().
      enforcingExpectationsWith(expectationEnforcer).
      with1Arg();
  }

  public void paint(Widget widget) {
    paint.call(widget);
  }
  
  public final MethodMocker1<Void, Widget> paint; 
  
}

File: scala/src/test/scala/sample2/MockWidgetPainter.scala
// Copyright 2011 Kiel Hodges
package sample2

import replicant._
import experiment1._

class MockWidgetPainter extends XMockController[WidgetPainter] { self =>

  val testDouble: WidgetPainter = new WidgetPainter {
    def paintWidget(widget: Widget) { self.paintWidget(widget).response }
  }

  val paintWidget = method("paintWidget", testDouble.paintWidget _)
  
}
